servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapSingleTest.java

```diff
@@ -515,31 +517,37 @@ public void onComplete() {
 
     @Test
     public void testEmitFromQueue() throws Exception {
-        List<LegacyTestSingle<Integer>> emittedSingles = new ArrayList<>();
-        LegacyBlockingSubscriber<Integer> subscriber = new LegacyBlockingSubscriber<>();
+        List<TestSingle<Integer>> emittedSingles = new ArrayList<>();
+        TestCollectingPublisherSubscriber<Integer> blockingSubscriber = new TestCollectingPublisherSubscriber<>();
+        TestPublisherSubscriber<Integer> subscriber = new TestPublisherSubscriber.Builder<Integer>()
+                .lastSubscriber(blockingSubscriber).build();
         toSource(source.flatMapMergeSingle(integer -> {
-            LegacyTestSingle<Integer> s = new LegacyTestSingle<>();
+            TestSingle<Integer> s = new TestSingle<>();
             emittedSingles.add(s);
             return s;
         }, 2)).subscribe(subscriber);
         subscriber.request(Long.MAX_VALUE);
         source.onNext(1, 1);
         assertThat("Unexpected number of Singles emitted.", emittedSingles, hasSize(2));
 
-        LegacyTestSingle<Integer> single1 = emittedSingles.remove(0);
-        LegacyTestSingle<Integer> single2 = emittedSingles.remove(0);
+        TestSingle<Integer> single1 = emittedSingles.remove(0);
+        TestSingle<Integer> single2 = emittedSingles.remove(0);
+
         executorService.execute(() -> {
             single1.onSuccess(2);
             single2.onSuccess(3);
         });
 
-        subscriber.awaitAndVerifyAwaitingItem(2); // Second item would only be sent once the first thread is free.
-        subscriber.unblock(2);
-        subscriber.verifyReceived(2);
+        Integer nextItem = blockingSubscriber.takeOnNext();
+        assertNotNull(nextItem);
+        assertEquals(2, nextItem.intValue());
+        nextItem = blockingSubscriber.takeOnNext();
+        assertNotNull(nextItem);
+        assertEquals(3, nextItem.intValue());
+        assertFalse(blockingSubscriber.pollTerminal(200, MILLISECONDS));
```

Comment:
Do we need this check? 

I am questioning usefullness more since we are adding `pollTerminal()` on `TestCollectingPublisherSubscriber` for this reason. Generally since we have test timeouts, we do not add APIs for time based polls, it will be better to avoid.

Reply:
This is to test the affirmative that a terminal event isn't delivered until we expect it to be. Just checking `blockingSubscriber.awaitOnComplete();` would be "sufficient" but the event may have been delivered for some reason other than the source completing (e.g. operator bug).